Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -347,6 +347,118 @@ zsh: # . ~/.zshrc
+
+```
+iPhone:~ root# lsof -i -a -p 1
+COMMAND PID USER   FD   TYPE             DEVICE SIZE/OFF NODE NAME
+launchd   1 root    2u  IPv6 0x69c2ce210efdba03      0t0  TCP localhost:mpm-flags (LISTEN)
+launchd   1 root    3u  IPv4 0x69c2ce21125bc873      0t0  TCP localhost:mpm-flags (LISTEN)
+launchd   1 root    8u  IPv6 0x69c2ce210efdcc63      0t0  TCP localhost:intu-ec-client (LISTEN)
+launchd   1 root    9u  IPv4 0x69c2ce210eeac873      0t0  TCP localhost:intu-ec-client (LISTEN)
+launchd   1 root   12u  IPv4 0x69c2ce210eeabebb      0t0  TCP localhost:socks (LISTEN)
+launchd   1 root   13u  IPv4 0x69c2ce210eeadbe3      0t0  TCP localhost:ansoft-lm-1 (LISTEN)
+launchd   1 root   14u  IPv6 0x69c2ce210efdc643      0t0  TCP *:62078 (LISTEN)
+launchd   1 root   15u  IPv4 0x69c2ce210eeae59b      0t0  TCP *:62078 (LISTEN)
+launchd   1 root   18u  IPv6 0x69c2ce210efdcc63      0t0  TCP localhost:intu-ec-client (LISTEN)
+launchd   1 root   19u  IPv4 0x69c2ce210eeac873      0t0  TCP localhost:intu-ec-client (LISTEN)
+launchd   1 root   20u  IPv6 0x69c2ce210efdba03      0t0  TCP localhost:mpm-flags (LISTEN)
+launchd   1 root   22u  IPv4 0x69c2ce210eeabebb      0t0  TCP localhost:socks (LISTEN)
+launchd   1 root   23u  IPv4 0x69c2ce210eeadbe3      0t0  TCP localhost:ansoft-lm-1 (LISTEN)
+launchd   1 root   24u  IPv4 0x69c2ce21125bc873      0t0  TCP localhost:mpm-flags (LISTEN)
+launchd   1 root   27u  IPv6 0x69c2ce210efdc023      0t0  TCP *:ssh (LISTEN)
+launchd   1 root   28u  IPv6 0x69c2ce210efdc023      0t0  TCP *:ssh (LISTEN)
+launchd   1 root   29u  IPv4 0x69c2ce210eeaef53      0t0  TCP *:ssh (LISTEN)
+launchd   1 root   30u  IPv4 0x69c2ce210eeaef53      0t0  TCP *:ssh (LISTEN)
+launchd   1 root   31u  IPv4 0x69c2ce211253b90b      0t0  TCP 192.168.1.12:ssh->192.168.1.8:62684 (ESTABLISHED)
+launchd   1 root   42u  IPv4 0x69c2ce211253b90b      0t0  TCP 192.168.1.12:ssh->192.168.1.8:62684 (ESTABLISHED)
+```
+
+##### Sandbox Inspection
+
+On iOS, each application gets a sandboxed folder to store its data. As per the iOS security model, an application's sandboxed folder cannot be accessed by another application. Additionally, the users do not have direct access to the iOS filesystem, thus preventing browsing or extraction of data from the filesystem. In the earlier versions of iOS there were certain application available which can be used to browse the device's filesystem, but in the recent version of iOS the sandboxing rules are more stringent and these applications do not work anymore. As a result, sandboxed data on recent iOS versions, sandboxed data can only be accessed on a jailbroken device. As part of jailbreaking process, the application sandbox protection is disabled/weakened and thus enabling the an easy access to sandboxed folders.
```

Comment:
```suggestion
On iOS, each application gets a sandboxed folder to store its data. As per the iOS security model, an application's sandboxed folder cannot be accessed by another application. Additionally, the users do not have direct access to the iOS filesystem, thus preventing browsing or extraction of data from the filesystem. In earlier versions of iOS there were applications available which can be used to browse the device's filesystem, but in the recent version of iOS the sandboxing rules are more stringent and these applications do not work anymore. As a result, if you need to access the filesystem it can only be accessed on a jailbroken device. As part of the jailbreaking process, the application sandbox protection is disabled and thus enabling an easy access to sandboxed folders.

If you want to only access the sandboxed data of an app, this can also be done on a non-jailbroken device by using objection and Frida. How this can be done is described in "[Accessing App Data Directories](0x06b-Basic-Security-Testing.md#accessing-app-data-directories)" in the chapter iOS Basic Security Testing. This chapter also gives an overview of the folder structure and which directories you should analyse. 
```

Reply:
Do you have further information to which version of iOS you are referring to? Otherwise it's a big vague, as we do not mention the app name neither the iOS versions.

Reply:
I was changing it, as the information is already described in great detail in our iOS basic security testing chapter, so it makes sense to link to this section, as otherwise we have redundant content. Hope that makes sense to you.
Please remove then line 414 till the end.

Reply:
You are correct, removed the content. Seems there are still unexplored corners for me :).

Reply:
No worries, it's hard to know over 500 pages inside out :-)